linode/resource_linode_nodebalancer_node.go

```diff
@@ -181,6 +181,18 @@ func resourceLinodeNodeBalancerNodeUpdate(d *schema.ResourceData, meta interface
 		return fmt.Errorf("Error parsing Linode NodeBalancer ID %v as int", d.Get("config_id"))
 	}
 
+	// If node doesn't exist, create a new one.
+	// This is necessary because the deletion of an instance
+	// will automatically delete its associated node.
+	_, err = client.GetNodeBalancerNode(context.Background(), nodebalancerID, configID, int(id))
```

Comment:
The canonical way recreating valid resources that are prone to be deleted through some side-effect is through the `read` function. 

Essentially the same structure but you wipe the id from the state, which will trigger recreation:
```
if linodeErr, ok := err.(*linodego.Error); ok && linodeErr.Code == 404 {
   d.SetId("")
    return nil
}
```

Reply:
If this doesn't work, I'd rather just mention that this is a known side effect of deleting a linode that has a nodebalancer dependency because creating in update just doesn't follow terraform design